packages/medusa/src/loaders/custom-api.ts

```diff
@@ -0,0 +1,56 @@
+import { Express, NextFunction, Request, Response, Router } from "express"
+import { ConfigModule } from "../types/global"
+
+enum AllowedMethods {
+  GET = "GET",
+  POST = "POST",
+  DELETE = "DELETE",
+}
+
+type Handler = (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => Promise<void>
+
+export type MedusaRoute = {
+  name: string
+  method: AllowedMethods
+  path: string
+  handler: Handler | Handler[]
+  routes: MedusaRoute[]
+  options: {}
+}
+
+export default ({
+  app,
+  configModule,
+}: {
+  app: Express
+  configModule: ConfigModule
+}) => {
+  configModule.routes.map(async (customRoute) => {
+    let { handler, path, method, routes, name } = customRoute
+
+    if (!method || !path || !handler || !name) {
```

Comment:
todo: As mentionned by phil, the error need to be managed at the appropriate level and should use the logged from loaders/logger instead of the console API.

here is what I suggest for managing the errors and the update for the handlers

```ts
configModule.routes.map(async (customRoute) => {
    const { handlers, path, method, routes, name } = customRoute

    if (routes) {
      const subRouter = Router({ mergeParams: true })

      app.use(path, subRouter)

      routes.map((route) => {
        const { handlers, path, method } = route
        if (!method || !path) {
          console.warn(`Invalid config for custom route: ${name}`)
        }

        subRouter[method.toLowerCase()](path, ...handlers)
      })
    } else {
      if (!method || !path || !handlers || !name) {
        console.warn(`Invalid config for custom route: ${name}`)
      }

      app[method.toLowerCase()](path, ...handlers)
    }
  })
```